Zf3.feature/api

pmattmann opened the pull request:
Camp, Period, Day - Service
API-Testcases
Swagger:   /api-docu

pmattmann committed:
Camp, Period, Day-Service + Tests

pmattmann committed:
Services, Tests, Swagger

check out:  /api-docu

pmattmann committed:
Merge branch 'devel' of https://github.com/ecamp/ecamp3 into zf3.feature/api

# Conflicts:
#	composer.json
#	composer.lock
#	module/eCampCore/src/EntityService/CampService.php
#	module/eCampCore/src/EntityService/PeriodService.php
#	module/eCampLib/src/Service/BaseService.php

pmattmann committed:
div. fixes

usu commented on module/eCampLib/test/PHPUnit/AbstractHttpControllerTestCase.php:
```diff
@@ -23,13 +24,46 @@ public function setUp() {
+        return $this->getEntityManager($name)->getRepository($entityName);
+    }
+
+    protected function getRandomEntity($entityName, $name = null) {
```
Das macht die Tests aber nicht gerade deterministisch. Wirklich eine gute Idee?

pmattmann replied:
Je nach Test spielt es keine Rolle, welche Entität du bekommst.
Wenn es jedoch für den Test relevant ist, dann sollte diese Methode natürlich nicht verwendet werden.

usu reviewed (commented):
Bezüglich Swagger: Wie würde man jetzt vorgehen, um die einzelnen Endpunkte zu dokumentieren (erlaubte Parameter, etc.)?

usu commented:
Bei Aufruf von http://localhost/api erhalte ich eine PHP Notice, wenn ich nicht authentifiziert bin. War vermutlich aber vorher schon so.

pmattmann committed:
Bugfix

PHP-Notice, wenn nicht eingeloggt.

pmattmann commented:
Bezüglich Swagger:
Dies ist definitiv erst eine erste Skizze. Im Moment kann man damit noch nicht wirklich dokumentieren.
Es zeigt viel mehr auf, welche Endpunkte existieren und welche HTTP-Verben dafür vorgesehen/implementiert sind.
Luft nach Oben ist da noch viel vorhanden ;)

pmattmann committed:
Merge branch 'devel' of https://github.com/ecamp/ecamp3 into zf3.feature/api

pmattmann committed:
API Entitäten + Swagger

CampType, Camp, EventCategory, Event
Swagger Doku:
@swaggerTitle, @swaggerBody, @swaggerPath, @swaggerQuery

pmattmann committed:
Api-Tests

EventType, EventTypeFactory, EventTypePlugin,
EventTemplate, EventTemplateContainer

pmattmann committed:
EventInstanceTest

pmattmann committed:
Split Backend-Frontend
